### DISCUSSION TITLE
archery集成ldap无法登陆

### DISCUSSION THREAD

**Discussion by tangtangli** [hhyo/Archery#1629] [Q&A] [answered]
archery配置ldap之后，浏览器提示用户名或密码错误，请重新输入，去ldap日志查看，用户已经认证成功。

archery服务报错如下：
[2022-06-16 14:00:03,221][MainThread:140070598219584][task_id:django_auth_ldap][backend.py:368][WARNING]- Caught LDAPError while authenticating lishuai: INVALID_CREDENTIALS({'msgtype': 97, 'msgid': 1, 'result': 49, 'desc': 'Invalid credentials', 'ctrls': [], 'info': '80090308: LdapErr: DSID-0C09041C, comment: AcceptSecurityContext error, data 52e, v4563'})
[2022-06-16 14:02:05,792][MainThread:140070598219584][task_id:django_auth_ldap][backend.py:368][WARNING]- Caught LDAPError while authenticating lishuai: INVALID_CREDENTIALS({'msgtype': 97, 'msgid': 1, 'result': 49, 'desc': 'Invalid credentials', 'ctrls': [], 'info': '80090308: LdapErr: DSID-0C09041C, comment: AcceptSecurityContext error, data 52e, v4563'})

配置如下：
    AUTH_LDAP_SERVER_URI = "ldap://172.16.66.xx"
    # ldap认证的另一种方式,使用时注释AUTH_LDAP_USER_DN_TEMPLATE

    AUTH_LDAP_BIND_DN = "cn=betadmin,ou=xx,dc=xx,dc=cn"
    AUTH_LDAP_BIND_PASSWORD = "xx"
    AUTH_LDAP_USER_SEARCH = LDAPSearch(
        'ou=xxx,dc=xxx,dc=cn',
        ldap.SCOPE_SUBTREE,
        '(uid=%(user)s)',)

    AUTH_LDAP_ALWAYS_UPDATE_USER = True  # 每次登录从ldap同步用户信息
    AUTH_LDAP_USER_ATTR_MAP = {  # key为archery.sql_users字段名，value为ldap中字段名，用户同步信息
        "username": "givenName",
        "display": "sn",
        "email": "mail"
    }
请帮看下。

**Comment by LeoQuote** [comment]
写的很清楚啊, 用户密码不对

**Reply by tangtangli** [reply]
> 写的很清楚啊, 用户密码不对

ldap里日志已经看到账户认证成功的日志了，这个web界面里还是报错。

**Reply by LeoQuote** [reply]
因为这个组件不是 archery 提供的, 是另一个第三方的组件, 建议你还是多搜搜看, 或者去这个项目的 issue 区再问问

**Reply by tangtangli** [reply]
> 因为这个组件不是 archery 提供的, 是另一个第三方的组件, 建议你还是多搜搜看, 或者去这个项目的 issue 区再问问

好的，多谢

**Comment by tangtangli** [comment, marked as answer]
已解决

**Reply by LeoQuote** [reply]
有空的话可以分享下解决的过程或者方法, 给后面的人一点经验, 感谢你的分享

**Reply by wanghuigithub** [reply]
遇到相同问题：LdapErr: DSID-0C0903D3, comment: AcceptSecurityContext error, data 52e, v3839'

**Reply by wanghuigithub** [reply]
我理解是ssl。想问下怎么关闭ssl

**Reply by tangtangli** [reply]
可以参考下配置文件。
    AUTH_LDAP_SERVER_URI = "ldap://172.16.66.xx"
    AUTH_LDAP_BIND_DN = "cn=admini,ou=xx,dc=xx,dc=cn"
    AUTH_LDAP_BIND_PASSWORD = "xxxx"
    AUTH_LDAP_USER_SEARCH = LDAPSearch("ou=xxx,dc=xx,dc=cn",ldap.SCOPE_SUBTREE,'(cn=%(user)s)',)
    AUTH_LDAP_ALWAYS_UPDATE_USER = True
    AUTH_LDAP_USER_ATTR_MAP = {
        "display": "displayName",
        'first_name': 'givenName',
        'email': 'mail',
    }

**Reply by Codecaver** [reply]
请教下，你们部署都是直接改的setting.py而不是在 .env里面配置变量值？

**Comment by tangtangli** [comment]
忘记了，好像是setting.py吧，这个产品不健壮，有各种问题感觉



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2023年5月9日(周二) 下午4:56
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [hhyo/Archery] archery集成ldap无法登陆 (Discussion #1629)




 
请教下，你们部署都是直接改的setting.py而不是在 .env里面配置变量值？
